workshop/mathIsFun/ComplexNumber.cpp

```diff
@@ -0,0 +1,165 @@
+*/
+void ComplexNumber::setImaginary(const double imaginary) noexcept {
+    m_imaginary = imaginary;
+}
+
+
+/*
+ * checks if complex numbers are equal
+ * @param other complex number to be compared with this
+ * @return true if both numbers are equal else false
+*/
+bool ComplexNumber::equal(const ComplexNumber& other) const noexcept {
+    return (this->m_real == other.m_real) && (this->m_imaginary == other.m_imaginary);
+}
+
+
+/*
+ * sum two complex numbers
+ * @param other complex number to be sum with this
+ * @return sum of this and other
+*/
+ComplexNumber ComplexNumber::add(const ComplexNumber& other) const noexcept {
+    return ComplexNumber(this->m_real + other.m_real, this->m_imaginary + other.m_imaginary);
+}
+
+
+/*
+ * substract two complex numbers
+ * @param other complex number to be substracted from this
+ * @return substraction of this with other
+*/
+ComplexNumber ComplexNumber::substract(const ComplexNumber& other) const noexcept {
+    return ComplexNumber(this->m_real - other.m_real, this->m_imaginary - other.m_imaginary);
+}
+
+
+/*
+ * multiply two complex numbers
+ * @param other complex number to be multipy with this
+ * @return multiply of this and other
+*/
+ComplexNumber ComplexNumber::multiply(const ComplexNumber& other) const noexcept {
+    return ComplexNumber(this->m_real * other.m_real - this->m_imaginary * other.m_imaginary,
+                         this->m_imaginary * other.m_imaginary + this->m_imaginary * other.m_real);
+}
+
+
+/*
+ * operator which checks if complex numbers are equal
+ * @param other complex number to be compared with this
+ * @return true if both numbers are equal else false
+*/
+bool ComplexNumber::operator==(const ComplexNumber& other) const noexcept {
+    return equal(other);
+}
+
+
+/*
+ * operator which sum two complex numbers
+ * @param other complex number to be sum with this
+ * @return sum of this and other
+*/
+ComplexNumber ComplexNumber::ComplexNumber::operator+(const ComplexNumber& other) const noexcept {
+    return add(other);
+}
+
+
+/*
+ * operator which substract two complex numbers
+ * @param other complex number to be substracted from this
+ * @return substraction of this with other
+*/
+ComplexNumber ComplexNumber::ComplexNumber::operator-(const ComplexNumber& other) const noexcept {
+    return substract(other);
+}
+
+
+/*
+ * operator which multiply two complex numbers
+ * @param other complex number to be multipy with this
+ * @return multiply of this and other
+*/
+ComplexNumber ComplexNumber::ComplexNumber::operator*(const ComplexNumber& other) const noexcept {
+    return multiply(other);
+}
+
+
+
+/*
+ * gets visual representation of complex number
+ * @param strm stream to store in the representation of the complex number
+ * @param num complex number to get representation of
+ * @return stream which represents the complex number
+*/
+std::ostream& operator<<(std::ostream& strm, const ComplexNumber& num) noexcept {
```

Comment:
I like what you've done here but this function can be both shortened, as well improved performance-wise.
For example - we don't need to create 3 strings if there's a chance we'll ignore all of them and print "0" to the screen...